executor/runtime/types/pod.go

```diff
@@ -0,0 +1,330 @@
+package types
+
+import (
+	"encoding/base64"
+	"regexp"
+	"time"
+
+	"github.com/Netflix/titus-executor/api/netflix/titus"
+	"github.com/Netflix/titus-executor/uploader"
+	vpcTypes "github.com/Netflix/titus-executor/vpc/types"
+	"github.com/golang/protobuf/proto" // nolint: staticcheck
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+)
+
+// PodContainer is an implementation of Container backed only by a kubernetes pod.
+// This is currently using the base64'ed ContainerInfo until all fields are ported over to annotations
+type PodContainer struct {
+	command       []string
+	entrypoint    []string
+	hostnameStyle string
+	ipv4Address   *string
+	pod           *corev1.Pod
+	titusInfo     *titus.ContainerInfo
+}
+
+func NewPodContainer(pod *corev1.Pod, ipv4Address *string) (*PodContainer, error) {
+	if pod == nil {
+		return nil, errors.New("missing pod")
+	}
+	if ipv4Address == nil {
+		return nil, errors.New("missing ipv4 address")
+	}
+
+	c := &PodContainer{
+		ipv4Address: ipv4Address,
+		pod:         pod,
+	}
+	if err := c.extractContainerInfoFromPod(); err != nil {
+		return nil, err
+	}
+
+	if val, ok := c.titusInfo.GetPassthroughAttributes()[hostnameStyleParam]; ok {
+		c.hostnameStyle = val
+	}
+
+	entrypoint, command, err := parseEntryPointAndCommand(c.titusInfo)
+	if err != nil {
+		return nil, err
+	}
+	if entrypoint != nil {
+		c.entrypoint = entrypoint
+	}
+	if command != nil {
+		c.command = command
+	}
+
+	return c, nil
+}
+
+func (c *PodContainer) extractContainerInfoFromPod() error {
+	str, ok := c.pod.GetAnnotations()["containerInfo"]
+	if !ok {
+		return errors.New("unable to find containerInfo annotation")
+	}
+
+	data, err := base64.StdEncoding.DecodeString(str)
+	if err != nil {
+		return errors.Wrap(err, "unable to base64 decode containerInfo annotation")
+	}
+
+	var cInfo titus.ContainerInfo
+	err = proto.Unmarshal(data, &cInfo)
+	if err != nil {
+		return errors.Wrap(err, "unable to decode containerInfo protobuf")
+	}
+
+	c.titusInfo = &cInfo
+	return nil
+}
+
+func (c *PodContainer) AllowCPUBursting() bool {
```

Comment:
Why is this feature and others below hard-coded to false or empty string or nil?

Reply:
This is only a partial implementation - the rest will be done as part of the "native pods" work.

Reply:
ok